File: rxbilling/src/main/java/com/vanniktech/rxbilling/PurchaseAble.java
package com.vanniktech.rxbilling;

import androidx.annotation.NonNull;

/** Something that can be purchased. Either an in-app product or a subscription. */
public interface PurchaseAble {
  /** @return The product ID for the product. */
  @NonNull String sku();

  /** @return Value must be inapp for an in-app product or subs for subscriptions. */
  @NonNull String type();
}

File: rxbilling/src/main/java/com/vanniktech/rxbilling/PurchaseAble.kt
package com.vanniktech.rxbilling

/** Something that can be purchased. Either an in-app product or a subscription.  */
interface PurchaseAble {
  /** The product ID for the product. */
  val sku: String

  /** Value must be inapp for an in-app product or subs for subscriptions. */
  val type: String
}
